### ISSUE TITLE
[java] AssignmentInOperand not raised when inside do-while loop

### ISSUE THREAD

**Issue by csrma** [pmd/pmd#5915]
**Affects PMD Version:** 7.15.0

**Rule:** [AssignmentInOperand](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#assignmentinoperand)

**Description:**
I think this rule is not activated when the `AssignmentInOperand` happens inside the `while` condition of a `do-while` loop. It is detected for all other cases (if, while). In the minimal reproducible example below, the error is only raised on line 9 only, although line 13 should be detected too. In the documentation, there are properties that can be set to allow this rule for if, for, while, but no mention if it's activated for `do-while` or not. If it's not supposed to support `do-while` loops, that should be mentioned as well.

**Code Sample demonstrating the issue:**

```java
class BuggyCode {
    public int getX() {
        return 4;
    }

    public int showBug() {
        int x = 2;
        do {
            if ((x = getX()) == 3){     // line - 9 => detected
              System.out.println(x);
            }
            x++;
        } while ((x = getX()) <= 3);    // line - 13 => not detected
        return x;
    }
}
```

**Expected outcome:**

PMD should report a violation at line 13, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
